### PR TITLE
7858 ux assurance changes for an RL registration.

### PR DESCRIPTION
**Pull request by doug-lovett**
Signed-off-by: Doug Lovett <doug@diamante.ca>

*Issue #:* /bcgov/entity#7858

*Description of changes:*
- Add validation of lienAmount
- Copy styling of date picker from bcrs shared components
- No general collateral for RL
- No MH vehicle type for RL vehicle collateral
- Change surrender date display format on step 1

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

File: ppr-ui/src/components/collateral/composables/useVehicle.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import { reactive, toRefs, computed } from '@vue/composition-api'
-import { VehicleTypes } from '@/resources'
+import { VehicleTypes, VehicleTypesNoMH } from '@/resources'
```
Comment (jordiwes) [inline ppr-ui/src/components/collateral/composables/useVehicle.ts:2]: What is VehicleTypesNoMH used for?

File: ppr-ui/tests/unit/RegistrationLengthTrust.spec.ts [hunk 2]
```diff
@@ -121,7 +121,6 @@ describe('RegistrationLengthTrust RL tests', () => {
     expect(wrapper.vm.showErrorLienAmount).toBe(false)
     expect(wrapper.vm.showErrorSurrenderDate).toBe(false)
     expect(wrapper.vm.minSurrenderDate).toBeDefined()
-    expect(wrapper.vm.maxSurrenderDate).toBeDefined()
```
Comment (jordiwes) [inline ppr-ui/tests/unit/RegistrationLengthTrust.spec.ts:124]: I wonder if we should pop in a little surrender date validation test while we are in here

File: ppr-ui/src/components/collateral/composables/useVehicle.ts (hunk 1 above)
Comment (doug-lovett) [reply ppr-ui/src/components/collateral/composables/useVehicle.ts:2]: A Repairer's Lien registration cannot have MH in the list.

Comment (jordiwes) [reply ppr-ui/src/components/collateral/composables/useVehicle.ts:2]: I don't see this resource actually used anywhere :)

Comment (doug-lovett) [reply ppr-ui/src/components/collateral/composables/useVehicle.ts:2]: EditCollateral.vue line 33

File: ppr-ui/tests/unit/RegistrationLengthTrust.spec.ts (hunk 2 above)
Comment (doug-lovett) [reply ppr-ui/tests/unit/RegistrationLengthTrust.spec.ts:124]: Unit test added in second commit.

File: ppr-ui/src/components/collateral/composables/useVehicle.ts (hunk 1 above)
Comment (jordiwes) [reply ppr-ui/src/components/collateral/composables/useVehicle.ts:2]: Gotcha! Sorry I missed that

File: ppr-ui/tests/unit/RegistrationLengthTrust.spec.ts (hunk 2 above)
Comment (jordiwes) [reply ppr-ui/tests/unit/RegistrationLengthTrust.spec.ts:124]: I was thinking something like 
wrapper.vm.$data.minSurrenderDate= '2012-08-01'
expect(wrapper.vm.lienAmountMessage).toBe('Bad date')

if you know what I mean... not to be picky :)

**Review by kialj876 — APPROVED** [review]
I like the idea of having a validation test for surrender date too, but looks good to me 👍

**Pull request merged**